suse_migration_services/migration_config.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2019 SUSE Linux LLC.  All rights reserved.
+#
+# This file is part of suse-migration-services.
+#
+# suse-migration-services is free software: you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# suse-migration-services is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with suse-migration-services. If not, see <http://www.gnu.org/licenses/>
+#
+import yaml
+
+# project
+from suse_migration_services.defaults import Defaults
+
+
+class MigrationConfig(object):
+    """
+    **Implements reading of migration configuration file:**
+
+    /etc/migration-config.yml
+
+    The migration config file lives inside of the migration
+    live image system. It is usually provided as a pre-setup
+    version as part of the live migration image build.
+    """
+    target_products = {
+        'ses': 'SLES/15.1/x86_64',
+        'sles': 'SLES/15/x86_64'
+    }
+
+    def __init__(self):
+        self.config_data = None
+        with open(Defaults.get_migration_config_file(), 'r') as config:
+            self.config_data = yaml.safe_load(config)
+
+    def get_migration_product(self):
+        """
+        Returns the full qualified product
+
+        The value returned is passed to the --product option in the
+        zypper migration call
+
+        migration_product: product_name
+        """
+        migration_product = self.config_data.get('migration_product', 'sles')
+        if migration_product in self.target_products:
+            migration_product = self.target_products[migration_product]
+
```

Comment:
This doesn't work this way.

Since we provide a default value to get() we know that is nothing is specified in the config we have an entry in target_products{}. If the user specifies a value in the config and specifies something we do not understand we have to exit and abort the migration.

As currently implemented if the user specifies `bar` as the migration product then `bar` would trickle onto the command line for `zypper migration` which would probably trigger an error and that OK. But hen we might as well dispense of the member test. This should simply be:

```
migration_target = self.config_data.get('migration_product', 'sles')
migration_product = self.target_products.get(migration_target, 'Fail')
```
"Fail" is now the default and will trigger a failure of migration. But we should probably do better:

```
migration_target = self.config_data.get('migration_product', 'sles')
migration_product = self.target_products.get(migration_target, 'Fail')
if migration_product == 'Fail':
    log.error('The configured migration product "%s" is not supported, aborting migration attempt' % migration_target)
```